vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/ApplicationActions.java

```diff
@@ -42,4 +44,33 @@ public void activateApp(String bundleId)
                 "Application with the bundle identifier '%s' is not installed on the device", bundleId);
         interactor.activateApp(bundleId);
     }
+
+    /**
+     * Terminate the application if it's running.
+     * @param bundleId bundle identifier of the application to terminate.
+     */
+    public void terminateApp(String bundleId)
+    {
+        InteractsWithApps interactor = webDriverProvider.getUnwrapped(InteractsWithApps.class);
+        ApplicationState appState = interactor.queryAppState(bundleId);
+        Validate.isTrue(appState != ApplicationState.NOT_INSTALLED && appState != ApplicationState.NOT_RUNNING,
+                "Application with the bundle identifier '%s' is not installed or not running on the device",
+                bundleId);
+        Validate.isTrue(interactor.terminateApp(bundleId),
+                "Application with the bundle identifier '%s' hasn't been successfully terminated", bundleId);
+    }
+
+    /**
+     * Reinstall the application.
+     * @param bundleId bundle identifier of the application to reinstall.
+     */
+    public void reinstallApplication(String bundleId)
+    {
+        InteractsWithApps interactor = webDriverProvider.getUnwrapped(InteractsWithApps.class);
+        HasCapabilities hasCapabilities = webDriverProvider.getUnwrapped(HasCapabilities.class);
+        String appPath = hasCapabilities.getCapabilities().getCapability("app").toString();
+        Validate.isTrue(interactor.removeApp(bundleId),
+                "Application with the bundle identifier '%s' hasn't been successfully removed", bundleId);
```

Comment:
Maybe something simpler like "Unable to remove mobille application with the bundle identifier '%s'"